{Containerapp} `az containerapp env update`: Support to update `--dapr-connection-string`. Hide `--dapr-instrumentation-key`

Greedygre opened the pull request:
**Related command**
<!--- Please provide the related command with az {command} if you can, so that we can quickly route to the related person to review. --->

**Description**<!--Mandatory-->
<!--Why this PR? What is changed? What is the effect? etc. A high-quality description can accelerate the review process.-->

**Testing Guide**
<!--Example commands with explanations.-->

**History Notes**
<!--If your PR is not customer-facing, use {Component Name} in the PR title. Otherwise, use [Component Name] to allow our pipeline to add the title as a history note. If you need multiple history notes or would like to overwrite the note from the PR title, please fill in the following templates.-->

[Containerapp] `az containerapp env create`: Hide `--dapr-instrumentation-key` parameter
[Containerapp] `az containerapp env update`: Support to update `--dapr-connection-string`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

yonzhan commented:
Thank you for your contribution! We will review the pull request and get back to you soon.

Juliehzl commented on src/azure-cli/azure/cli/command_modules/containerapp/_params.py:
```diff
@@ -160,6 +160,10 @@ def load_arguments(self, _):
         c.argument('dapr_connection_string', options_list=['--dapr-connection-string', '-d'], help='Application Insights connection string used by Dapr to export service to service communication telemetry.')
 
+    with self.argument_context('containerapp env update', arg_group='Dapr') as c:
+        c.argument('instrumentation_key', options_list=['--dapr-instrumentation-key'], help='Application Insights instrumentation key used by Dapr to export Service to Service communication telemetry. Use "none" to remove it.')
```
why do we need to add --dapr-instrumentation-key?

Greedygre replied:
Updated, we should not add --dapr-instrumentation-key, also, we need to hide it because this argument is in deprecate process

yanzhudd commented:
> History Notes
> 
> [Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change
> [Component Name 2] az command b: Add some customer-facing feature

Since there are multiple changes in this PR, please split the changelog into individual items and list them in the History Notes section

Greedygre commented:
> > History Notes
> > [Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change
> > [Component Name 2] az command b: Add some customer-facing feature
> 
> Since there are multiple changes in this PR, please split the changelog into individual items and list them in the History Notes section

Addressed comment
